### PR TITLE
Add next step sections to starter readme

### PR DESCRIPTION
**Pull request by weikanglim**
The next step sections cover the journey to getting the application up to Azure.

### CODE REVIEW

File: templates/starter/bicep/README.md [hunk 1]
```diff
@@ -7,3 +7,65 @@ The following assets have been provided:
 - Infrastructure-as-code (IaC) files under the `infra` folder that demonstrate how to provision resources and setup resource tagging for `azd`.
 - A [dev container](https://containers.dev) configuration file under the `.devcontainer` directory that installs infrastructure tooling by default. This can be readily used to create cloud-hosted developer environments such as [GitHub Codespaces](https://aka.ms/codespaces).
 - Continuous deployment workflows for CI providers such as GitHub Actions under the `.github` directory, and Azure Pipelines under the `.azdo` directory that work for most use-cases.
+
+## Next Steps
+
+### Step 1: Add application code
+
+Initialize the service source code projects anywhere under the current directory. Ensure that all source code projects can be built successfully.
```
Comment (savannahostrowski) [inline templates/starter/bicep/README.md:15]: Might be nice to number steps under each header for easier parsing.

File: templates/starter/bicep/README.md [hunk 2]
```diff
@@ -7,3 +7,65 @@ The following assets have been provided:
...
+
+### Step 2: Provision Azure resources
+
+Update or add Infrastructure as Code (IaC) files to provision the relevant Azure resources. This can be done incrementally, as the list of [Azure resources](https://learn.microsoft.com/en-us/azure/?product=popular) are explored and added. As an example, a reference library that contains all of the Bicep modules used by the azd templates can be found [here](https://github.com/Azure-Samples/todo-nodejs-mongo/tree/main/infra/core). All Azure resources available in Bicep format can be found [here](https://learn.microsoft.com/en-us/azure/templates/).
+
+Run `azd provision` whenever you want to ensure that changes made are applied correctly and work as expected.
+
+### Step 3: Tie in application and infrastructure
+
+There are certain pieces required that help tie-in application and infrastructure together. These are pieces that help expose your application to other Azure resources, or help allow your application to take advantage of Azure capabilities. For more details, see [below](#additional-details). This includes the following:
```
Comment (savannahostrowski) [inline templates/starter/bicep/README.md:31]: Nit but "pieces" of what? Maybe there's more a descriptive way to phrase this?

Comment (puicchan) [inline templates/starter/bicep/README.md:31]:
```suggestion
There are certain pieces required that help tie-in application and infrastructure together. These are pieces that help expose your application to other Azure resources, or help allow your application to take advantage of Azure capabilities. For more details, see [Additional Details](#additional-details). This includes the following:
```

File: templates/starter/bicep/README.md [hunk 3]
```diff
@@ -7,3 +7,65 @@ The following assets have been provided:
 - Infrastructure-as-code (IaC) files under the `infra` folder that demonstrate how to provision resources and setup resource tagging for `azd`.
 - A [dev container](https://containers.dev) configuration file under the `.devcontainer` directory that installs infrastructure tooling by default. This can be readily used to create cloud-hosted developer environments such as [GitHub Codespaces](https://aka.ms/codespaces).
 - Continuous deployment workflows for CI providers such as GitHub Actions under the `.github` directory, and Azure Pipelines under the `.azdo` directory that work for most use-cases.
+
+## Next Steps
+
```
Comment (savannahostrowski) [inline templates/starter/bicep/README.md:12]: Can we link to the additional details section here?

**Review by puicchan — APPROVED** [review]

File: templates/starter/bicep/README.md [hunk 4]
```diff
@@ -7,3 +7,65 @@ The following assets have been provided:
...
+Run `azd provision` whenever you want to ensure that changes made are applied correctly and work as expected.
+
+### Step 3: Tie in application and infrastructure
+
+There are certain pieces required that help tie-in application and infrastructure together. These are pieces that help expose your application to other Azure resources, or help allow your application to take advantage of Azure capabilities. For more details, see [below](#additional-details). This includes the following:
+
+1. Set up application settings for the code running in Azure to reference other Azure resources.
+1. Set up managed identities to allow the code running in Azure to access other Azure resources.
```
Comment (savannahostrowski) [inline templates/starter/bicep/README.md:34]: Can we link to the section below around managed identities ( I think this is possible with some special syntax)? We start to introduce concepts without defining them.

**Review by savannahostrowski — CHANGES_REQUESTED** [review]
Left some comments on Bicep, which probably also apply to the TF readme as well

File: templates/starter/bicep/README.md (hunk 2 above)
Comment (weikanglim) [reply templates/starter/bicep/README.md:31]: @puicchan Do you have some inspiration here based on how users generally think about "tie in"? I can definitely reword this, but I just don't know if what I'm writing resonates with users.

Comment (puicchan) [reply templates/starter/bicep/README.md:31]: How about 
```suggestion
Configurations in Bicep help to tie application and infrastructure together. These are pieces that help expose your application to other Azure resources, or help allow your application to take advantage of Azure capabilities. Some examples:
```

File: templates/starter/bicep/README.md [hunk 5]
```diff
@@ -7,3 +7,65 @@ The following assets have been provided:
...
+1. If you are using Key Vault, set up required secrets in KeyVault that then can be referenced by your application settings, with the use of managed identities.
+1. Configure settings on the hosting platform to match your application's needs. This may include networking, security options, or more advanced configuration that helps you take advantage of Azure capabilities.
+
```
Comment (puicchan) [inline templates/starter/bicep/README.md:37]:
```suggestion
For more details, see [Additional Details](#additional-details). 
```

File: templates/starter/bicep/README.md (hunk 1 above)
Comment (weikanglim) [reply templates/starter/bicep/README.md:15]: Let me know if you have specific improvements so I can incorporate. I haven't quite figured out how to massage the other sections to a consistent numbering format.

Comment (weikanglim) [reply templates/starter/bicep/README.md]: I made some changes along these lines, it does read better

File: templates/starter/bicep/README.md (hunk 3 above)
Comment (weikanglim) [reply templates/starter/bicep/README.md:12]: Let me know if the current changes still need work

**Review by gkulin — APPROVED** [review]

**Review by savannahostrowski — APPROVED** [review]

**Pull request merged**